src/messages/message.ts

```diff
@@ -1,39 +1,116 @@
-import { ChainType } from "../accounts/account";
+/**
+ * Chain defines which account was used to publish a message.
+ * It is automatically provided when publishing messages.
+ *
+ * Warning: Avax, CSDK, NEO are currently not supported by the TS sdk.
+ */
+import { ProgramContent } from "./program/programModel";
+
+export enum Chain {
+    DOT = "DOT",
+    ETH = "ETH",
+    SOL = "SOL",
+    NULS = "NULS",
+    NULS2 = "NULS2",
+    AVAX = "AVAX",
+    CSDK = "CSDK",
+    NEO = "NEO",
+}
+
+/**
+ * Supported hash functions
+ */
+export enum HashType {
+    sha256 = "sha256",
+}
+
+/**
+ * Message types supported by Aleph
+ *
+ * Warning: Program is currently not supported by the TS sdk.
+ */
+export enum MessageType {
+    post = "POST",
+    aggregate = "AGGREGATE",
+    store = "STORE",
+    program = "PROGRAM",
+    forget = "FORGET",
+}
+
+export enum ItemType {
+    inline = "inline",
+    storage = "storage",
+    ipfs = "ipfs",
```

Comment:
Are you merging a storage engine and a storage method ?

Reply:
As far as I know nop, I updated this structure to fit with [aleph-message](https://github.com/aleph-im/aleph-message/blob/main/aleph_message/models/__init__.py)